flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -931,11 +940,17 @@ private void onTriggerFailure(
                     abortPendingCheckpoint(checkpoint, cause);
                 }
             } else {
-                LOG.info(
-                        "Failed to trigger checkpoint for job {} because {}",
+                LOG.warn(
+                        "Failed to trigger checkpoint for job {}. ({} consecutive failed attempts so far)",
                         job,
-                        throwable.getMessage());
+                        numUnsuccessful,
+                        throwable);
 
+                if (checkpoint == null
+                        && CheckpointFailureReason.IO_EXCEPTION
+                                == cause.getCheckpointFailureReason()) {
+                    statsTracker.reportFailedCheckpointsWithoutInProgress();
```

Comment:
I don't really agree with this condition. I mean if flink fails when the pending checkpoint has been created already then we update statistics independently on the reason for fail(even if the consecutive fail counter won't be increased). But if we have not created the pending checkpoint then we check for IO_EXCEPTION. In my opinion, it is not consistent behavior.

And in general, I don't really like how we use `CheckpointStatsCounts`. Right now we increase the checkpoint in progress count and the total checkpoint number only when the pending checkpoint has been created. But if we fail before the pending checkpoint we need to call the separated method(as it was created in this PR) to handle this situation. 
Maybe it is better to have logic like that:
- increment the number of inProgressCheckpoints and the total checkpoints as the first step before even checkpoint id has been created.
- add the PendingCheckpoint to stats tracker as soon as it would be created(right now it happens at the same time with previous step after the PendingCheckpoint is created)
- decrement the total number of checkpoints and checkpoints in progress as the last step of the checkpoint(the same as now)
- decrement checkpoints in progress but increment failed checkpoint number if the checkpoint fails(the same as now for PendingCheckpoint but use it for non existing PendingCheckpoint as well)

Important notice that the number of failed checkpoint in statistic and continuousFailureCounter in CheckpointFailureManager are totally different numbers. The first one can be grown all the time while the second one can be always zero if the reason of fail doesn't consider as critical.

@1996fanrui , @pnowojski , WDYT?